td-shim/src/bin/td-shim/main.rs

```diff
@@ -274,28 +272,38 @@ fn boot_builtin_payload(
     };
 }
 
-// We reserved two pages for the IDT used before payload setup
-// its own one.
-pub fn switch_idt(idt_base: u64) {
-    let idt_page = unsafe {
-        memslice::get_dynamic_mem_slice_mut(memslice::SliceType::RelocatedIdt, idt_base as usize)
-    };
+// Reuse mailbox for the IDT used before payload setup its own one.
+pub fn switch_idt(mailbox_base: u64) {
+    // Put the runtime IDT at the firmware reserved area inside the mailbox page
+    let idt_offset = 0xC00;
+    let idt_base = mailbox_base + idt_offset as u64;
+    let idt_page = &mut unsafe {
+        memslice::get_dynamic_mem_slice_mut(
+            memslice::SliceType::RelocatedMailbox,
+            mailbox_base as usize,
+        )
+    }[idt_offset..];
 
-    let mut idt = unsafe { &mut *(idt_page.as_ptr() as u64 as *mut Idt) };
+    // Set IDT only for interrupt [0..32)
+    let mut idt =
+        unsafe { core::slice::from_raw_parts_mut(idt_page.as_ptr() as u64 as *mut IdtEntry, 32) };
 
-    let handler_offset = size_of::<Idt>();
+    let handler_offset = size_of::<IdtEntry>() * 32;
     let handler_size = empty_exception_handler_size();
```

Comment:
How do you guarantee the handler_size is small enough and not exceed the firmware region?